File: day21/Linked-List-Cycle.py
class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head is None or head.next is None:
            return False
        
        slow = head
        fast = head.next
        
        while fast is not None and fast.next is not None:
            if slow == fast:
                return True
            
            slow = slow.next
            fast = fast.next.next
            
        return False

File: day21/Linked-List-Cycle.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct ListNode {
    int val;
    struct ListNode *next;
};

bool hasCycle(struct Linked-List-Cycle)
{
    if (head == NULL || head->next == NULL) {
        return false;
    }

    struct ListNode *slow = head;
    struct ListNode *fast = head->next;
    
    while (fast != NULL && fast->next != NULL) {
        if (slow == fast) {
            return true; 
        }
        slow = slow->next;
        fast = fast->next->next;
    }
    return false;   
    
};
